src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/FileFields.scala

```diff
@@ -1,25 +1,46 @@
 package uk.gov.nationalarchives.tdr.api.graphql.fields
 
+import java.time.{LocalDateTime, ZonedDateTime}
 import java.util.UUID
 
 import io.circe.generic.auto._
 import sangria.macros.derive._
 import sangria.marshalling.circe._
-import sangria.schema.{Argument, Field, InputObjectType, ObjectType, fields}
-import uk.gov.nationalarchives.tdr.api.auth.{ValidateHasExportAccess, ValidateUserHasAccessToConsignment}
+import sangria.schema.{Argument, BooleanType, Field, InputObjectType, ObjectType, OptionType, fields}
+import uk.gov.nationalarchives.tdr.api.auth.{ValidateHasExportAccess, ValidateUserHasAccessToConsignment, ValidateUserOwnsFiles}
 import uk.gov.nationalarchives.tdr.api.consignmentstatevalidation.ValidateNoPreviousUploadForConsignment
-import uk.gov.nationalarchives.tdr.api.graphql.ConsignmentApiContext
+import uk.gov.nationalarchives.tdr.api.graphql.{ConsignmentApiContext, DeferFileMetadata}
 import uk.gov.nationalarchives.tdr.api.graphql.validation.UserOwnsConsignment
 import uk.gov.nationalarchives.tdr.api.graphql.fields.FieldTypes._
+import uk.gov.nationalarchives.tdr.api.service.FileMetadataService.FileMetadataValues
 
 object FileFields {
   case class Files(fileIds: Seq[UUID])
+  case class FileDetails(fileId: UUID, consignmentId: UUID, userId: UUID, dateTime: LocalDateTime, checksumMatches: Option[Boolean])
```

Comment:
Is there any reason you can't use the existing `File` type? https://github.com/nationalarchives/tdr-consignment-api/blob/master/src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/ConsignmentFields.scala#L49

I think that would mean you'd only have one `File` type in the schema, whether you fetched `{ file { id } }` or `{ consignment { file { id } } }`.

Reply:
That `file` type does not contain all the information that could be retrieved about the file:

- consignmentId
- userId
- dateTime
- checksumMatches

Could potentially update the existing type to include to include this, but this will all have to be optional fields so as not to break existing code.

Reply:
Ah, so there are fields when we could return `None` just because we haven't implemented them? That doesn't seem great... Maybe we should have a chat about how we're approaching our GraphQL schema design?